src/bvals/comms/bnd_info.cpp

```diff
@@ -152,9 +170,9 @@ int GetBufferSize(std::shared_ptr<MeshBlock> pmb, const NeighborBlock &nb,
   const int isize = cb.ie(in) - cb.is(in) + 2;
   const int jsize = cb.je(in) - cb.js(in) + 2;
   const int ksize = cb.ke(in) - cb.ks(in) + 2;
-  return (nb.ni.ox1 == 0 ? isize : Globals::nghost) *
-         (nb.ni.ox2 == 0 ? jsize : Globals::nghost) *
-         (nb.ni.ox3 == 0 ? ksize : Globals::nghost) * v->GetDim(6) * v->GetDim(5) *
+  return (nb.ni.ox1 == 0 ? isize : Globals::nghost + 1) *
+         (nb.ni.ox2 == 0 ? jsize : Globals::nghost + 1) *
+         (nb.ni.ox3 == 0 ? ksize : Globals::nghost + 1) * v->GetDim(6) * v->GetDim(5) *
```

Comment:
Is the buffer size increased to account for send/recv data that is (not) owned?

Reply:
Yes, I have been not very careful about calculating buffer sizes. I essentially just take the maximum size possible (which corresponds to assuming a nodal field is being communicated). This means that the buffers are sometimes bigger than they need to be, which doesn't pose any issues for correctness. It does mean that the memory footprint is maybe a little larger than it needs to be, but this does mean that all of the buffer arrays can come from the same three memory pools.